Benchmarks/Benchmarks/SwiftKafkaBenchmarks/Utilities.swift

```diff
@@ -0,0 +1,293 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Ideally we would deduplicate this code but I know it is hard since we are in a separate Swift Package here.
@FranzBusch what do you think? I was thinking maybe vend these functions as [`@__spi`](https://github.com/apple/swift/blob/main/docs/ReferenceGuides/UnderscoredAttributes.md#_spispiname) in the `Kafka` target?

Reply:
Yeah I think we can expose some of the internal RD abstractions as SPI and use it here directly.

Reply:
@blindspotbounty let me know if I can help you here

Reply:
Sounds good to me as well. If I got it right:
1. Put topic creation/deletion functions to Kafka target as `internal` + mark them as `@_spi`
2. Import Kafka in Tests and Benchmark with `@_spi(Internal) import Kafka` 

Please, let me know if that's not what you keep in mind.

Reply:
This is what I have imagined!

Reply:
We can probably also do the same for `func createTestMessages` and `func sendAndAcknowledgeMessages`

Reply:
Thank you for clarification!

Regarding the latter I had some thoughts in mind about adding new target lib `KafkaTestUtils` that would be imported by either any tests or benchmarks without modifiers. 
For example in `Kafka` target:
```
/// RDKafkaClient+TopicCreation.swift
extension RDKafkaClient {
    @_spi(Internal) public func _createUniqueTopic(timeout: Int32) throws -> String
    @_spi(Internal) public func _deleteTopic(_ topic: String, timeout: Int32) throws
}
/// KafkaConsumerConfiguration.swift
extension KafkaConsumerConfiguration {
    @_spi(Internal) public var dictionary: [String: String]
}
```
Or even mark only some variables as `@_spi`:
```
final class RDKafkaClient: Sendable {
    @_spi(Private) public let kafkaHandle: OpaquePointer
}
```
while in `KafkaTestUtils`:
```
@testable @_spi(Internal) import Kafka
...
/// RDKafkaClient+TopicCreation.swift
extension RDKafkaClient {
    public func createUniqueTopic(timeout: Int32) throws -> String {
    ...
    }
    public func deleteTopic(_ topic: String, timeout: Int32) throws {
    ...
    }
    @discardableResult
    func withKafkaHandlePointer<T>(_ body: (OpaquePointer) throws -> T) rethrows -> T {
    ...
    }
}
/// KafkaConsumerConfiguration.swift
extension KafkaConsumerConfiguration {
    public var rdKafkaConfig: [String: String] { self.dictionary }
}
```

Then in Benchmark or in Test, we import `KafkaTestUtils` (+ optionally `Kafka`) and use both interfaces without modifiers while not polishing production code base. 
Though it might be a slight overengineering  :)